src/lib/holocene/input.svelte

```diff
@@ -0,0 +1,108 @@
+<script lang="ts">
+  import Icon from '$lib/holocene/icon/index.svelte';
+  import type { IconName } from '$lib/holocene/icon/paths';
+
+  export let id: string;
+  export let label: string;
+  export let value: string;
+  export let icon: IconName = '';
+  export let placeholder = label;
+  export let name = id;
+  export let copyable: boolean = false;
+  export let theme: 'dark' | 'light' = 'light';
+
+  let copied = false;
+  const handleCopy = () =>
+    navigator.clipboard
+      .writeText(value)
+      .then(() => {
+        copied = !copied;
+        setTimeout(() => (copied = false), 1000);
+      })
+      .catch((error) => console.error(error));
```

Comment:
This code is copied across the app a few times, is there a better way to share singleton stateful utilities across components in svelte à la custom hooks in react?

Reply:
This looks promising https://svelte.dev/repl/7580c4426c1947d8aa3d149a05bdc895?version=3.38.3